src/Compilers/CSharp/Portable/Emitter/Model/AttributeDataAdapter.cs

```diff
@@ -137,15 +138,15 @@ private MetadataCreateArray CreateMetadataArray(TypedConstant argument, EmitCont
 
         private static MetadataTypeOf CreateType(TypedConstant argument, EmitContext context)
         {
-            Debug.Assert(argument.Value != null);
+            Debug.Assert(argument.ValueInternal != null);
             var moduleBeingBuilt = (PEModuleBuilder)context.Module;
             var syntaxNodeOpt = (CSharpSyntaxNode)context.SyntaxNodeOpt;
             var diagnostics = context.Diagnostics;
-            return new MetadataTypeOf(moduleBeingBuilt.Translate((TypeSymbol)argument.Value, syntaxNodeOpt, diagnostics),
-                                      moduleBeingBuilt.Translate((TypeSymbol)argument.Type, syntaxNodeOpt, diagnostics));
+            return new MetadataTypeOf(moduleBeingBuilt.Translate((TypeSymbol)argument.ValueInternal, syntaxNodeOpt, diagnostics),
+                                      moduleBeingBuilt.Translate((TypeSymbol)argument.TypeInternal, syntaxNodeOpt, diagnostics));
```

Comment:
>TypeInternal [](start = 86, length = 12)

`InternalType`, `InternalValue` would read better.

Reply:
> InternalType, InternalValue would read better.

I think that would make them less discoverable. This way "Value" and "ValueInternal" are next to each other in the completion list. 

---
In reply to: [340383880](https://github.com/dotnet/roslyn/pull/39498#discussion_r340383880) [](ancestors = 340383880)